robusta/src/main.rs

```diff
@@ -271,33 +279,48 @@ async fn main() {
     update_bindings();
 
     info!("Starting server");
-    let (send, recv) = tokio::sync::mpsc::channel(100);
+    let (send, recv) = tokio::sync::mpsc::channel(10000);
     let state = load_state(send.clone());
 
-    if *FETCH_TRAINS {
-        kvv::init().await;
-
-        // fetch departures every 60 seconds and send them to the game logic queue
-        tokio::spawn(async move {
-            let mut interval = tokio::time::interval(Duration::from_secs(60));
-            loop {
-                interval.tick().await;
-                let departures = kvv::fetch_departures_for_region().await;
-                if departures.is_empty() {
-                    warn!("Fetched no departures");
-                }
-                if let Err(err) = send
-                    .send(InputMessage::Server(ServerMessage::Departures(departures)))
-                    .await
-                {
-                    error!("Error while fetching data: {err}")
+    let fetch_trains = if *FETCH_TRAINS {
+        Box::pin(async move {
+            kvv::init().await;
+            // fetch departures every 60 seconds and send them to the game logic queue
+            tokio::spawn(async move {
+                let mut interval = tokio::time::interval(Duration::from_millis(100));
+                let mut timeout = tokio::time::interval(Duration::from_secs(10));
+                loop {
+                    interval.tick().await;
+                    // Skip fetching updates if no player is currently connected
+                    if !PLAYERS_ONLINE.load(std::sync::atomic::Ordering::Relaxed) {
+                        continue;
+                    }
+
+                    let departures = kvv::fetch_departures_for_region().await;
+                    if departures.is_empty() {
+                        warn!("Fetched no departures");
+                    }
+                    if let Err(err) = send
+                        .send(InputMessage::Server(ServerMessage::Departures(departures)))
+                        .await
+                    {
+                        error!("Error while fetching data: {err}")
+                    }
+                    timeout.tick().await;
                 }
-            }
-        });
-    }
+            });
+        }) as Pin<Box<dyn Future<Output = ()>>>
+    } else {
+        Box::pin(std::future::ready(())) as Pin<Box<dyn Future<Output = ()>>>
+    };
 
+    pin!(fetch_trains);
```

Comment:
Isn't `fetch_train` already pinned?

Reply:
Ah, that was left over from before I needed to type erase them